### ISSUE TITLE
[Tests] Test Suite Improvements

### ISSUE THREAD

**Issue by compleatang** [monax/cli#863]
**Problem** -- we're relying on images which are production level images in order to run our gotests. In addition we need to move towards cleaner separation of the tests which interact with docker images/containers and those that only manipulate files. 

**Ideas** -- the following are my suggestions.
## ToolTests Job
- [ ] produce a dummy `quay.io/eris/test` container that can be ran as a daemon, logs a bit, and just runs forever. CLI team should build it. This should be kept in `common` and built and updated via the common pipeline maintained by operations team.
- [ ] move any container interaction to use solely `test` or `data` images. within the gotests we should have this fully isolated so that no images we _actually_ use by actual users are utilized during the gotests sequence.
- [ ] systematize the service definitions used throughout the gotests sequence. compile these into the binary and write them during `eris init --testing` to `$HOME/.eris/scratch/test-$UUID` (or something like that). we need a bit more sophisticated writing capabilities here; cloning from github should be a nonstarter for this test sequence.
- [ ] ensure that `eris init --testing`  only pulls `test` and `data` images rather than all the things.
- [ ] eventually we will want to separate out those tests which interact with containers and those that do not, but this is the lowest item on the checklist
- [ ] unify the ERIS_ROOT dir divergence. some of the tool tests (go tests) are dropping files into /tmp/eris and some of them are dropping files into $HOME/.eris. Neither is particularly great assumption. We should be able to give the initialization sequence a single directory which all the tests fixtures will run from and utilize. In general, however, it is kind of a hassle for users that we _insist_ on this. If we have to unify that pipeline for testing purposes, then we should expose that as a globalConfig setting that defaults to $HOME/.eris but also respects an env var. I would suggest we split these into `ERIS_ROOT` and `ERIS_TEST_ROOT`.
- [ ] the gotests clean sequence should remove the test/data images as well as any containers it started. right now there is overkill on the testTearDown sequence which limits our ability to run the tests in parallel.
## StackTests Job
- [ ] should just use `eris init` without the testing flag.
- [ ] should not build containers, but rather pull from quay.io

**Comment by zramsay** [comment]
- [ ] refactor https://github.com/eris-ltd/eris-cli/blob/master/tests/README.md

**Comment by zramsay** [comment]
handled by specific issues
